buildSrc/src/main/kotlin/otel.japicmp-conventions.gradle.kts

```diff
@@ -52,9 +44,56 @@ class AllowNewAbstractMethodOnAutovalueClasses : AbstractRecordingSeenMembers()
 }
 
 class SourceIncompatibleRule : AbstractRecordingSeenMembers() {
+
+  fun ignoreAddLogRecordProcessorCustomizerReturnTypeChange(member: JApiCompatibility): Violation? {
+    if (member is JApiClass &&
+      member.newClass.get().name.equals("io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder") &&
+      member.isChangeCausedByClassElement
+    ) {
+      // member.isChangeCausedByClassElement check above
+      // limits source of changes to fields, methods and constructors
+
+      for (method in member.methods) {
+        if (!method.isSourceCompatible()) {
+          // addLogRecordProcessorCustomizer method had a return type change from base to impl class,
+          // japicmp (correctly) doesn't consider it as a METHOD_RETURN_TYPE_CHANGED
+          // compatibility issue. But still thinks that something wrong.
+          // Since it thinks that method did not change, it reports it as class-level violation,
+          // and we need to suppress it, but keep other checks on.
+          if (method.name.equals("addLogRecordProcessorCustomizer") &&
+            method.compatibilityChanges.isEmpty() &&
+            method.changeStatus == JApiChangeStatus.UNCHANGED) {
+            return null;
+          }
+          return Violation.error(method, "Method is not source compatible: $method")
```

Comment:
adding `member.toString()` provides much better diagnostics

before:
`Class io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder: Not source compatible`

after
`Class io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder: Not source compatible: JApiClass [fullyQualifiedName=io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder, changeStatus=MODIFIED, compatibilityChanges=[]]`


or 

before
`	Method io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.addLogRecordProcessorCustomizer(java.util.function.BiFunction): Method is not source compatible`

after

`	Method io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.addLogRecordProcessorCustomizer(java.util.function.BiFunction): Method is not source compatible. JApiMethod [oldMethod=io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.addLogRecordProcessorCustomizer(java.util.function.BiFunction), newMethod=io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder.addLogRecordProcessorCustomizer(java.util.function.BiFunction), returnType=JApiReturnType [oldReturnTypeOptional=io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer, newReturnTypeOptional=io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer, changeStatus=UNCHANGED], getCompatibilityChanges()=[]]`